### ISSUE TITLE
Map generation sometimes fails

### ISSUE THREAD

**Issue by DanielVF** [Lux-AI-Challenge/Lux-Design-2021#79]
```
TypeError: Cannot read property 'hasResource' of undefined
    at Object.generateGame (/opt/homebrew/lib/node_modules/@lux-ai/2021-challenge/lib/es5/Game/gen.js:151:37)
    at Function.<anonymous> (/opt/homebrew/lib/node_modules/@lux-ai/2021-challenge/lib/es5/logic.js:87:38)
    at step (/opt/homebrew/lib/node_modules/@lux-ai/2021-challenge/lib/es5/logic.js:33:23)
    at Object.next (/opt/homebrew/lib/node_modules/@lux-ai/2021-challenge/lib/es5/logic.js:14:53)
    at /opt/homebrew/lib/node_modules/@lux-ai/2021-challenge/lib/es5/logic.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/opt/homebrew/lib/node_modules/@lux-ai/2021-challenge/lib/es5/logic.js:4:12)
    at Function.LuxDesignLogic.initialize (/opt/homebrew/lib/node_modules/@lux-ai/2021-challenge/lib/es5/logic.js:60:16)
    at LuxDesign.<anonymous> (/opt/homebrew/lib/node_modules/@lux-ai/2021-challenge/lib/es5/design.js:67:62)
    at step (/opt/homebrew/lib/node_modules/@lux-ai/2021-challenge/lib/es5/design.js:48:23)
    
```

**Comment by DanielVF** [comment]
Also sometimes (with different numbers as the error):

```
TypeError: Cannot read property '6' of undefined
    at GameMap.getCell (/opt/homebrew/lib/node_modules/@lux-ai/2021-challenge/lib/es5/GameMap/index.js:44:27)
    at Object.generateGame (/opt/homebrew/lib/node_modules/@lux-ai/2021-challenge/lib/es5/Game/gen.js:151:22)
    at Function.<anonymous> (/opt/homebrew/lib/node_modules/@lux-ai/2021-challenge/lib/es5/logic.js:87:38)
    at step (/opt/homebrew/lib/node_modules/@lux-ai/2021-challenge/lib/es5/logic.js:33:23)
    at Object.next (/opt/homebrew/lib/node_modules/@lux-ai/2021-challenge/lib/es5/logic.js:14:53)
    at /opt/homebrew/lib/node_modules/@lux-ai/2021-challenge/lib/es5/logic.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/opt/homebrew/lib/node_modules/@lux-ai/2021-challenge/lib/es5/logic.js:4:12)
    at Function.LuxDesignLogic.initialize (/opt/homebrew/lib/node_modules/@lux-ai/2021-challenge/lib/es5/logic.js:60:16)
    at LuxDesign.<anonymous> (/opt/homebrew/lib/node_modules/@lux-ai/2021-challenge/lib/es5/design.js:67:62)```

**Comment by StoneT2000** [comment]
hm wait i thought this was fixed, will check this out.

**Comment by StoneT2000** [comment]
probs fixed in #51 now
